vaadin-date-time-picker-flow-parent/vaadin-date-time-picker-flow/src/main/java/com/vaadin/flow/component/datetimepicker/DateTimePicker.java

```diff
@@ -391,6 +396,123 @@ public String getLabel() {
         return getElement().getProperty("label");
     }
 
+    /**
+     * Set the aria-label of the component to the given text.
+     *
+     * @param ariaLabel
+     *            the aria-label text to set or {@code null} to clear
+     */
+    public void setAriaLabel(String ariaLabel) {
+        getElement().setProperty("accessibleName", ariaLabel);
+    }
+
+    /**
+     * Gets the aria-label of the component.
+     *
+     * @return an optional aria-label of the component or an empty optional if
+     *         no aria-label has been set
+     */
+    public Optional<String> getAriaLabel() {
+        return Optional.ofNullable(getElement().getProperty("accessibleName"));
+    }
+
+    /**
+     * Contains DateTimePicker internalization properties
+     */
+    public static class DateTimePickerI18n implements Serializable {
+        private String dateLabel;
+        private String timeLabel;
+
+        public String getDateLabel() {
+            return dateLabel;
+        }
+
+        public void setDateLabel(String dateLabel) {
+            this.dateLabel = dateLabel;
+        }
+
+        public String getTimeLabel() {
+            return timeLabel;
+        }
+
+        public void setTimeLabel(String timeLabel) {
+            this.timeLabel = timeLabel;
+        }
+    }
+
+    /**
+     * Sets the accessible name for the date picker.
+     * <p>
+     * The final value is a concatenation of the accessible name from
+     * DateTimePicker's {@link #getAriaLabel()} or {@link #getLabel()} and the
+     * given accessible name.
+     *
+     * @param dateLabel
+     *            the value to be used as part of date picker aria-label.
```

Comment:
```suggestion
     * Sets the accessible label for the date picker.
     * <p>
     * The final value is a concatenation of the accessible label from
     * DateTimePicker's {@link #getAriaLabel()} or {@link #getLabel()} and the
     * given accessible label.
     *
     * @param dateLabel
     *            the value to be used as part of date picker aria-label.
```

Reply:
"The accessible label" seems to be a better option since the web component also uses this naming in JSDoc.